### DISCUSSION TITLE
HMR on html files - possible?

### DISCUSSION THREAD

**Discussion by TechAkayy** [ElMassimo/jekyll-vite#11] [General]
Hi there,

Thanks for this amazing repo. 

I notice that when I try updating docs/_frontend/components/NavBarActions.vue, vite HMR works. But when I update docs/_layouts/post.html (.html file), HMR doesn't work, and rather it reloads the full site. 

Is this expected behaviour? Is it possible to have HMR when working on the layout files? 

Thanks bunch!

**Comment by ElMassimo** [comment]
Hi!

HMR needs to be supported by each framework, so it wouldn't be possible to HMR layout changes given that Jekyll doesn't support that.

Full page reloads are the expected behavior.

**Reply by TechAkayy** [reply]
Thanks, makes sense. That would probably explain why it's wise choice to bet on single-file-components (.vue) in Iles, and why astro sticked to a html-like format (.astro) as well, both supporting HMR.
